src/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,55 @@
+import {
+  createContext,
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useState,
+  type FC,
+  type ReactNode,
+} from 'react';
+import type { User } from '../core/interfaces';
+
+interface AuthContextState {
+  user: User | null;
+  setUser: (user: User | null) => void;
+}
+
+interface AuthProviderProps {
+  children: ReactNode;
+}
+
+const initialState: AuthContextState = {
+  user: null,
+  setUser: () => {},
+};
+
+export const AuthContext = createContext<AuthContextState>(initialState);
+
+export const AuthProvider: FC<AuthProviderProps> = ({ children }) => {
+  const [user, setUserContext] = useState<User | null>(null);
```

Comment:
context слово можно убрать из названия стейта

Reply:
ты имеешь ввиду setUserContext > setUser? У меня далее по коду будет перекликаться название функции обновления состояния с функцией, которую передаем в качестве значения в контексте. Я хотел более чистое название передавать в значение контекста.
```suggestion
  const setUser = useCallback((user: User | null) => {
    setUserContext(user);
  }, []);
  
  return (
    <AuthContext.Provider value={{ user, setUser }}>
      {children}
    </AuthContext.Provider>
  );
```